### DISCUSSION TITLE
How to obtain peer_addr in UdpSocket?

### DISCUSSION THREAD

**Discussion by PragmaTwice** [tokio-rs/tokio#3309] [Q&A] [answered]
I know that `tokio::net::UdpSocket` can be stateless, but a peer_addr should be stored in `UdpSocket` after `UdpSocket::connect` is called (even `std::net::UdpSocket` has a `UdpSocket::peer_addr`).
I do not want to store `(UdpSocket, SocketAddr)` into my map because the `SocketAddr` should be stored inside `UdpSocket`.

**Comment by taiki-e** [comment, marked as answer]
This will be added in #4362.

**Reply by JamesHallowell** [reply]
Hey @taiki-e, it looks like #4362 was merged and then reverted. Is it going to be added back in at some point? Thanks.

**Reply by Noah-Kennedy** [reply]
I believe this was reverted due to a revert to an earlier version of mio, so this should come back soon.

**Reply by taiki-e** [reply]
re-added in https://github.com/tokio-rs/tokio/pull/4611
